stdlib/public/Platform/android.modulemap

```diff
@@ -0,0 +1,676 @@
+  header "complex.h"
+  export *
+}
+module _ctype [system] {
+  header "ctype.h"
+  export *
+}
+module _errno [system] {
+  header "errno.h"
+  export *
+}
+module _fenv [system] {
+  header "fenv.h"
+  export *
+}
+// Use 'no_undeclared_includes' to avoid pulling in the libc++ headers.
+// FIXME: this can be dropped once NDK's libc++ 'std' module is split
+// into individual submodules.
+module _inttypes [system] [no_undeclared_includes] {
+  header "inttypes.h"
+  export *
+  use _stdint
+  use _sys_cdefs
+}
+module _limits [system] {
+  header "limits.h"
+  export *
+  explicit module posix_limits {
+    header "bits/posix_limits.h"
+    export *
+  }
+}
+module _locale [system] {
+  header "locale.h"
+  export *
+}
+module _malloc [system] {
+  header "malloc.h"
+  export *
+}
+module _math [system] {
+  header "math.h"
+  export *
+  link "m"
+}
+module _pty [system] {
+  header "pty.h"
+  export *
+}
+module _setjmp [system] {
+  header "setjmp.h"
+  export *
+}
+module _signal [system] {
+  header "signal.h"
+  // The 'signal.h' header unfortunately has a circular include
+  // with 'sys/ucontext.h' and thus it must be part of this module.
+  header "sys/ucontext.h"
+  export *
+  explicit module bits_signaltypes {
+    header "bits/signal_types.h"
+    export *
+  }
+}
+module _stdatomic [system] {
+  header "stdatomic.h"
+  export *
+}
+module _stdint [system] {
+  header "stdint.h"
+  export *
+}
+module _stdio [system] {
+  header "stdio.h"
+  export *
+  module stdio_ext {
+    header "stdio_ext.h"
+    export *
+  }
+}
+module _stdlib [system] {
+  header "stdlib.h"
+  export *
+}
+module _string [system] {
+  header "string.h"
+  export *
+}
+module _time [system] {
+  header "time.h"
+  export *
+  module sys_time {
+    header "sys/time.h"
+    export *
+  }
+  module sys_times {
+    header "sys/times.h"
+    export *
+  }
+}
+
+// POSIX and another android NDK headers.
+module alloca [system] {
+  header "alloca.h"
+  export *
+}
+
+module ar [system] {
+  header "ar.h"
+  export *
+}
+
+module cpio [system] {
+  header "cpio.h"
+  export *
+}
+
+module posix_filesystem [system] {
+  // This module groups all file, paths and filesystem
+  // operations into one module.
+  module dirent {
+    header "dirent.h"
+    export *
+  }
+  module fcntl {
+    header "fcntl.h"
+    export *
+    explicit module sys_fcntl {
+      header "sys/fcntl.h"
+      export *
+    }
+  }
+  module fnmatch {
+    header "fnmatch.h"
+    export *
+  }
+  module fts {
+    header "fts.h"
+    export *
+  }
+  module glob {
+    header "glob.h"
+    export *
+  }
+  module mntent {
+    header "mntent.h"
+    export *
+  }
+  module libgen {
+    header "libgen.h"
+    export *
+  }
+  module nl_types {
+    header "nl_types.h"
+    export *
+  }
+  module paths {
+    header "paths.h"
+    export *
+  }
+  module poll {
+    header "poll.h"
+    export *
+  }
+  module pwd {
+    header "pwd.h"
+    export *
+  }
+  module unistd {
```

Comment:
Is unistd really file system? I think of it for fork/exec and things like that. Although it is a little bit of a grab bag...

Reply:
yeah it's more of a grab bag, in this case it has a lot of read / write operations. But it probably would be better to move it out still.